lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStartHandler.java

```diff
@@ -31,38 +34,68 @@
 import java.util.logging.Logger;
 
 /**
- * This is bound to the start event of the clock and dispatches it to the approriate application code.
+ * This is bound to the start event of the clock and dispatches it to the appropriate application code.
  */
 public final class DriverStartHandler implements EventHandler<StartTime> {
   private static final Logger LOG = Logger.getLogger(DriverStartHandler.class.getName());
 
   private final Set<EventHandler<StartTime>> startHandlers;
   private final Optional<Set<EventHandler<StartTime>>> restartHandlers;
-  private final DriverStatusManager driverStatusManager;
+  private final Optional<Set<EventHandler<StartTime>>> serviceRestartHandlers;
+  private final Optional<DriverRestartManager> driverRestartManager;
 
   @Inject
   DriverStartHandler(@Parameter(org.apache.reef.driver.parameters.DriverStartHandler.class)
```

Comment:
I don't think you need the constructor for the case of application level restart handlers, but no service level restart handlers. Tang will inject an empty set by default for those, even if nothing is bound. Hence, writing the code under the assumption to always have the set of service-bound restart handlers is safe, and probably easier.